pkg/monitor/monitorapi/types.go

```diff
@@ -60,20 +60,135 @@ func EventLevelFromString(s string) (EventLevel, error) {
 }
 
 type Condition struct {
-	Level EventLevel
+	Level ConditionLevel
 
-	Locator string
-	Message string
+	// TODO: Goal here is to drop Locator/Message, and rename the structured variants to Locator/Message
+	Locator           string
+	StructuredLocator Locator
+	Message           string
+	StructuredMessage Message
 }
 
-type EventInterval struct {
+type LocatorType string
```

Comment:
huh, you or me?  This is a good idea, though we should kill "other" at some future point.